### PR TITLE
MSB-23 add material-ui

### PR DESCRIPTION
**Pull request by PavelEvleev**
add form, navigation

### CODE REVIEW

File: web/src/components/AddAuthor/index.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleAddClick = this.handleAddClick.bind(this);
+
+    }
+
+    handleChange = (event) =>{
+        this.setState({name: event.target.value});
+    }
+
```
Comment (PavelEvleev) [inline web/src/components/AddAuthor/index.jsx]: How I can hide form and show for example dialog "succes"?

Comment (lehaSVV2009) [reply web/src/components/AddAuthor/index.jsx]: I see 2 options:
1) Redirect `form save` to another route. Like you have `/books` and `/books/create` in `<Routes>...</Routes>`. `/books` - show all books. `/books/create` - Create single book. When you click `Create book`, it shows only create book form. When you click `Save`, it redirects to `/books` and show notification `Success` that disappears in 3 seconds.
2) Do everything on one route and do smth like `this.setState({showNotification: true})` when book is saved

File: web/package.json [hunk 2]
```diff
@@ -1,84 +1,83 @@
...
-    "moduleNameMapper": {
-      "\\.(jpe?g|png|gif|eot|otf|webp|svg|ttf|woff2?|mp[34]|webm|wav|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
-      "\\.(css|less|s[ac]ss|styl)$": "<rootDir>/__mocks__/styleMock.js",
-      "app": "<rootDir>/src/app.jsx"
+    "name": "MyReadedBooks",
+    "version": "1.0.0",
+    "description": "The app stores books that have been already read by user",
+    "repository": {
+        "type": "git",
+        "url": "https://github.com/PavelEvleev/MyReadedBooks"
```
Comment (lehaSVV2009) [inline web/package.json]: url другой теперь

Comment (lehaSVV2009) [inline web/package.json]: url поменялся

File: web/src/components/AddAuthor/index.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    handleAddClick = () =>{
```
Comment (lehaSVV2009) [inline web/src/components/AddAuthor/index.jsx]: Usually view and logic are located in different classes. e.g. 
```
handleAddClick = () => {
  this.props.onSaveAuthor(this.state.author)
}
```

File: web/src/components/AddAuthor/index.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        console.log(this.state.name);
+      api.fetchAddAuthor({name: this.state.name})
```
Comment (lehaSVV2009) [inline web/src/components/AddAuthor/index.jsx]: `api.createAuthor`

File: web/src/components/AddBook/index.jsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                    value={this.state.value}
+                    onChange={this.handleSelectedChange}
+                    >
+                    {
+                        this.state.authors.map(
+                            (author, index)=>
+                                <MenuItem key={index} value={author.id} primaryText={author.name}/>
+                            )
+                    }
+                </SelectField>
```
Comment (PavelEvleev) [inline web/src/components/AddBook/index.jsx]: How I can dynamically adding SelectField, and add selected item of each SelectField to common array, and subscribe at common event?
If we didn't find needed author, we can add him manual and fields must adding dynemically if it necesory for new authors.

File: api/src/main/java/app/services/UserService.java [hunk 6]
```diff
@@ -57,11 +57,8 @@ public void delete(Long id) {
...
     public void patch(Long id, Long bookId) {
        User user = userRepository.findOne(id);
-       Book toDelete = user.getBooks().stream().filter((book)->book.getId() == bookId).findAny().orElse(null);
-       if(toDelete!=null){
-           user.getBooks().remove(toDelete);
-       }
-       userRepository.save(user);
+       user.setBooks(user.getBooks().stream().filter(book -> !book.getId().equals(bookId)).collect(Collectors.toList()));
+       User u = userRepository.saveAndFlush(user);
     }
```
Comment (PavelEvleev) [inline api/src/main/java/app/services/UserService.java]: Didn't update date in DB!

File: api/src/main/java/app/services/UserService.java [hunk 7]
```diff
@@ -51,6 +57,22 @@ public void delete(Long id) {
         userRepository.delete(id);
     }
 
+
+    @Transactional
+    public void patch(Long id, Long bookId) {
+       User user = userRepository.findOne(id);
+       user.setBooks(user.getBooks().stream().filter(book -> !book.getId().equals(bookId)).collect(Collectors.toList()));
+       User u = userRepository.saveAndFlush(user);
```
Comment (lehaSVV2009) [inline api/src/main/java/app/services/UserService.java]: return

File: web/src/components/Home/DrawerOpenRightExample.jsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+  handleUsersClick = ()=>{
+    this.props.history.push('/users');
+    this.handleTouchTap();
+  }
+  
+  render() {
+    return (
+      <div>
+      <Drawer width={200} open={this.props.open} docked={false} onRequestChange={() =>this.handleTouchTap()} >
```
Comment (lehaSVV2009) [inline web/src/components/Home/DrawerOpenRightExample.jsx]: this.handleTouchTap

File: web/src/components/User/index.jsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+        return arBooks;
+    } 
+
+    handleAddClick = ()=>{
+
+        if(this.state.arraySelectedAuthors!=null && this.state.arraySelectedAuthors!=undefined){
+            api.addBooksToUser({userId: this.state.user.id, ids: this.state.arraySelectedAuthors})
+            .then((response)=>{
+                this.setState({user: response.data});
+                debugger;
```
Comment (lehaSVV2009) [inline web/src/components/User/index.jsx]: ...

**Pull request merged**